src/scripts/main.js

```diff
@@ -1,3 +1,102 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('.field');
+const rowRemove = document.querySelector('.remove-row');
+const rowAdd = document.querySelector('.append-row');
+const columnAdd = document.querySelector('.append-column');
+const columnRemove = document.querySelector('.remove-column');
+
+document.addEventListener('click', (e) => {
+  const target = e.target.closest('.button');
+
+  if (!target) {
+    return;
+  }
+
+  changeTable(target);
+});
+
+function changeTable(elem) {
+  const addRow = elem.closest('.append-row');
+  const addColumn = elem.closest('.append-column');
+  const removeRow = elem.closest('.remove-row');
+  const removeColumn = elem.closest('.remove-column');
+
+  switch (elem) {
+    case addRow:
+      const cloneElem = table.rows[0].cloneNode(true);
+
+      table.prepend(cloneElem);
+
+      if (table.rows.length === 10) {
+        addAttribute(addRow);
+      }
+      break;
+
+    case removeRow:
+      table.rows[table.rows.length - 1].remove();
+
+      if (table.rows.length === 2) {
+        addAttribute(removeRow);
+      }
+
+      break;
+
+    case addColumn:
+      [...table.rows].forEach((el) => {
+        const td = document.createElement('td');
+
+        el.prepend(td);
+      });
+
+      if (table.rows[0].cells.length === 10) {
+        addAttribute(addColumn);
+      }
+
+      break;
+
+    case removeColumn:
+      [...table.rows].forEach((el) => {
+        el.cells[el.cells.length - 1].remove();
+      });
+
+      if (table.rows[0].cells.length === 2) {
```

Comment:
The condition 'table.rows[0].cells.length === 2' is used to disable the removeColumn button. Ensure that this logic aligns with the requirements for the minimum number of columns.